public/app/features/variables/textbox/TextBoxVariablePicker.tsx

```diff
@@ -27,7 +27,15 @@ export function TextBoxVariablePicker({ variable }: Props): ReactElement {
         toVariablePayload({ id: variable.id, type: variable.type }, { propName: 'query', propValue: updatedValue })
       )
     );
-    variableAdapters.get(variable.type).updateOptions(variable);
+
+    if (onVariableChange) {
+      onVariableChange({
+        ...variable,
+        current: { ...variable.current, value: updatedValue },
+      });
+    } else {
+      variableAdapters.get(variable.type).updateOptions(variable);
+    }
```

Comment:
Nit: maybe this is clearer:
```suggestion
    if (onVariableChange) {
      onVariableChange({
        ...variable,
        current: { ...variable.current, value: updatedValue },
      });
      return;
    } 
    
    variableAdapters.get(variable.type).updateOptions(variable);
```